starcheck/src/lib/Ska/Parse_CM_File.pm

```diff
@@ -103,17 +93,33 @@ sub dither {
 	    }
           }
 	$dith_hist_fh->close();
-    
-    my @ok = grep { $dh_time[$_] < $bs_arr->[0]->{time} } (0 .. $#dh_time);
-    my @state = (@dh_state[@ok], @bs_state);
-    my @time   = (@dh_time[@ok], @bs_time);
-    my @params = (@dh_params[@ok], @bs_params);
+
+    # Confirm that last state matches kadi continuity ENAB/DISA
+    if ($kadi_dither->{'dither'} ne $dh_state[-1]){
+        print STDERR "Dither status in kadi commands does not match DITHER history\n";
+        return (1, undef);
```

Comment:
I there's something very weird, everything will end up getting checked with (20, 20) dither and a red warning about dither.

Reply:
This isn't really an additional check on kadi dynamic states, it is more of a check that the "cut" version from continuity is all as expected.  The defaults really aren't bad either if we end up with a rapid replan and an out-of-date kadi or some such... though I don't know if we want to be able to provide a state manually.